### PR TITLE
<filesystem>: Avoid additional allocations in path::lexically_relative

### LINKED ISSUE #937: <filesystem>: Avoid additional allocations in path::lexically_relative
See: https://github.com/microsoft/STL/pull/923#discussion_r447107800

#923 is implementing LWG-3070 (#333) by iterating through `_Path.relative_path()`. This can allocate memory for each element (which is also a `path`). Avoiding that would improve performance, but we need to carefully verify that directly inspecting the native string has the same behavior.

### PR DESCRIPTION
**Pull request by DailyShana**
fix #937

basically copied from https://github.com/microsoft/STL/pull/923/files#r447107800
I read the implement of related code and I think the behavior is the same.

### CODE REVIEW

File: stl/inc/filesystem [hunk 1]
```diff
@@ -1684,11 +1684,21 @@ namespace filesystem {
...
     }
 
+    _NODISCARD inline bool _Starts_with_root_name(const wchar_t* const _First, const wchar_t* const _Last) {
+        return _Find_root_name_end(_First, _Last) != _First;
+    }
+
     _NODISCARD inline bool _Relative_path_contains_root_name(const path& _Path) {
-        for (const auto& _File_name : _Path.relative_path()) {
-            if (!_Parse_root_name(_File_name.native()).empty()) {
+        auto& _Native    = _Path.native();
```
Comment (strega-nil-ms) [inline stl/inc/filesystem:1692]:
```suggestion
        const auto& _Native = _Path.native();
```

doesn't change the actual type, but I prefer explicitly marking `const` here.

**Review by strega-nil-ms — APPROVED** [review]
LGTM! Minor nit.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for improving the performance of the C++17 filesystem library! :open_file_folder: :rocket: :smile_cat:

**Pull request merged**